### PR TITLE
Adding keepalive action step to end of live runs to stop GitHub automatically disabling the workflow

### PR DESCRIPTION
**Pull request by MartyFox**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-16002

### Change description ###
Adding keepalive action step to end of live runs to stop GitHub automatically disabling the workflow

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: .github/workflows/reset-workflow-timeout.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Prevent scheduled jobs being disabled after 60 days
+# see https://github.community/t/no-notification-workflow-disabled-after-60-days/182169
+# existing GitHub notifications don't seem to work at all
+# this should trigger a dummy commit to the default branch every 50 days
+jobs:
+  keepalive:
+    name: Keep scheduled jobs running
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (timja) [inline .github/workflows/reset-workflow-timeout.yaml:16]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/reset-workflow-timeout.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      - uses: gautamkrishnar/keepalive-workflow@1.0.7
```
Comment (timja) [inline .github/workflows/reset-workflow-timeout.yaml:17]:
```suggestion
      - uses: gautamkrishnar/keepalive-workflow@2.0.6
```

File: .github/workflows/reset-workflow-timeout.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Keep scheduled jobs running
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+
+# Prevent scheduled jobs being disabled after 60 days
+# see https://github.community/t/no-notification-workflow-disabled-after-60-days/182169
+# existing GitHub notifications don't seem to work at all
+# this should trigger a dummy commit to the default branch every 50 days
```
Comment (timja) [inline .github/workflows/reset-workflow-timeout.yaml:10]: looks like v2 uses the API to keep it alive in someway by default and doesn't need to commit

File: .github/workflows/reset-workflow-timeout.yaml (hunk 1 above)
Comment (MartyFox) [reply .github/workflows/reset-workflow-timeout.yaml]: updated both actions to latest versions

File: .github/workflows/reset-workflow-timeout.yaml (hunk 3 above)
Comment (MartyFox) [reply .github/workflows/reset-workflow-timeout.yaml:10]: Updated to v2 as it was in the original change.

**Pull request merged**